assets/app/scripts/directives/nav.js

```diff
@@ -58,10 +58,11 @@ angular.module('openshiftConsole')
 
           sortedProjects = $filter('orderByDisplayName')(projects);
           options = _.map(sortedProjects, function(item) {
+            // truncates @ 35 as this prevents words from clobbering the caret
             return $('<option>')
-                      .attr("value", item.metadata.name)
+                      .attr('value', $filter('truncate')(item.metadata.name, 35))
```

Comment:
So this works as far as linking goes:
```javascript
return $('<option>')
        .attr("value", item.metadata.name)
        .attr("selected", item.metadata.name === name)
        .text($filter('truncate')($filter("uniqueDisplayName")(item, sortedProjects), 35));
```
Not sure about accessibility/screen reader issues yet, as inspecting the source everything I can see shows truncated text anyway. :confused: